src/NServiceBus.Core/Pipeline/Incoming/ReceivePipelineCompleted.cs

```diff
@@ -0,0 +1,40 @@
+namespace NServiceBus
+{
+    using System;
+    using Transport;
+
+    /// <summary>
+    /// The ReceivePipeline completed event.
```

Comment:
I don't think this is correct, the instance specific queue is using the mainPipelineExecutor as well so the event would fire there as well?

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/StartableEndpoint.cs#L145